### PR TITLE
Prevent stale-green-ci munger from blocking mungegithub when jenkins is down.

### LINKED ISSUE #3744: mungegithub: should be able to apply approved label if jenkins is down
I don't have access to the mungegithub logs, but it appears jenkins is down, and it appears the mungegithub submit queue loop frequency dropped to 0, or slowed way down.

http://velodrome.k8s.io/dashboard/db/monitoring?panelId=4&fullscreen&from=1501153226291&to=1501174852958

/area mungegithub
/kind bug

### PR DESCRIPTION
**Pull request by cjwagner**
fixes #3744 
/cc @rmmh @spxtr

### CODE REVIEW

**Comment by rmmh** [comment]
If jobs aren't starting, nothing will get through the queue either way. How does this materially change things?

File: mungegithub/mungers/stale-green-ci.go [hunk 1]
```diff
@@ -105,15 +117,26 @@ func (s *StaleGreenCI) Munge(obj *github.MungeObject) {
 				glog.Errorf("Failed to write retrigger old test comment")
 				return
 			}
-			ok := obj.WaitForPending(requiredContexts)
```
Comment (rmmh) [inline mungegithub/mungers/stale-green-ci.go]: Why not just not wait for pending here?

Comment (cjwagner) [inline mungegithub/mungers/stale-green-ci.go]: If we munged the same PR again before the context switches to pending we would make a duplicate comment in that case.

**Comment by cjwagner** [comment]
> If jobs aren't starting, nothing will get through the queue either way. How does this materially change things?

It doesn't unblock the queue at all, if jenkins is down, the queue is going to have problems. This just allows the other mungers to continue working properly such as applying the approved label.

**Pull request merged**